delivery_auto_refresh/models/sale_order.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# Copyright 2021 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    available_carrier_ids = fields.Many2many(
+        comodel_name="delivery.carrier",
+        compute="_compute_available_carrier_ids",
+    )
+
+    @api.onchange("partner_id", "partner_shipping_id")
+    def _onchange_partner_id(self):
+        if hasattr(super(), "_onchange_partner_id"):
+            return super()._onchange_partner_id()
+        default_carrier = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("delivery_auto_refresh.set_default_carrier")
+        )
+        if default_carrier:
+            for order in self:
+                action = order.action_open_delivery_wizard()
+                carrier_id = self.env["delivery.carrier"].browse(
+                    action["context"]["default_carrier_id"]
+                )
+                # If the carrier isn't allowed for the current shipping address, we wont
+                # default to it. In that case we'd try to fallback to the former carrier.
+                order.carrier_id = fields.first(
+                    (carrier_id | order.carrier_id).filtered(
+                        lambda x: x in order.available_carrier_ids._origin
+                    )
+                )
+
+    @api.depends("partner_shipping_id")
+    def _compute_available_carrier_ids(self):
+        """We want to apply the same carriers filter in the header as in the wizard"""
+        for sale in self:
+            wizard = self.env["choose.delivery.carrier"].new({"order_id": sale.id})
+            sale.available_carrier_ids = wizard.available_carrier_ids._origin
+
+    def _get_param_auto_add_delivery_line(self):
+        # When we have the context 'website_id' it means that we are doing the order from
+        # e-commerce. So we don't want to add the delivery line automatically.
+        if self.env.context.get("website_id"):
+            return False
+        auto_delivery = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("delivery_auto_refresh.auto_add_delivery_line")
+        )
+        return auto_delivery
```

Comment:
There's no need to assign a variable that we're immediately returning back
```suggestion
        return = (
            self.env["ir.config_parameter"]
            .sudo()
            .get_param("delivery_auto_refresh.auto_add_delivery_line")
        )
```